scripts/Makefile.build

```diff
@@ -291,7 +291,9 @@ $(obj)/%.lst: $(src)/%.c FORCE
 # Compile Rust sources (.rs)
 # ---------------------------------------------------------------------------
 
-rustc_cross_flags := --target=$(KBUILD_RUSTC_TARGET)
+# Need to use absolute path here; otherwise rustdoc and rustc compute
+# different hash for the target.
+rustc_cross_flags := --target=$(realpath $(KBUILD_RUSTC_TARGET))
```

Comment:
Rustc uses `path.canonicalize()` on the target spec location, which resolves symlinks. The reason this `realpath` has to be here at all is because rustdoc forgot to use `path.canonicalize()` too.

Reply:
Thanks! Then perhaps we can keep it for the moment, and remove it later on when your PR lands and we move to newer nightlies.

Reply:
We should probably mention the need to resolve the symlinks in the comment above to avoid forgetting it.

Reply:
Done

Reply:
https://github.com/rust-lang/rust/pull/85361 has been merged. The fix will be available on the next nightly.